main_test.go

```diff
@@ -458,3 +460,23 @@ func TestPortOverride(t *testing.T) {
 		t.Errorf("expected listen %s got %s", expected, listen)
 	}
 }
+
+func TestStatsWriteSuccess(t *testing.T) {
+	t.Parallel()
+	req, err := http.NewRequest("GET", "https://foo.bar", nil)
+	if err != nil {
+		t.Errorf("Error setting up request.")
+	}
+	ag.statsWriteSuccess(req, "test")
```

Comment:
Ideally, the `stats` object on `ag` would be some sort of mock or other thing that we can inspect afterwards. Or at the very least, we need to ensure that it was called appropriately.